app/models/sorn_xml_parser.rb

```diff
@@ -1,95 +1,127 @@
-    # an example of why paragraphs only won't work
-    # https://www.federalregister.gov/documents/full_text/xml/2020/10/13/2020-22534.xml
-    content = []
-    @parser.within('SUPLINF').each do |node|
-      if node.name != 'PRIACT' && node.name != 'SIG'
-        # skip section title
-        if node.name == 'HD' && node.include?('SUPPLEMENTARY')
-          next
-        else
-          content << node
-        end
+  def find_section(header)
+    # Get a named section of the PRIACT tag
+    # header of 'NUMBER' will match the section with key 'System Name and Number'
+    matched_header = @sections.keys.find{ |key| key.upcase.include? header }
+    @sections[matched_header]
+  end
+
+  private
+
+  def get_sections
+    # Gather the named sections of the PRIACT tag
+    sections = {}
+    current_header = nil
+    @parser.within('PRIACT').each do |node|
+      if node.name == 'HD'
+        current_header = cleanup_xml_element_to_string(node)
+        sections[current_header] = []
+      elsif current_header.nil?
+        next
+      elsif node.name == 'P'
+        # append cleaned strings
+        sections[current_header] << cleanup_xml_element_to_string(node)
       end
     end
-    content
+
+    # discard the rare nil keys
+    sections.except!(nil)
+    # Join array into a large string of paragraphs.
+    sections.transform_values!{|values| values.map{|p| "<p>#{p}</p>" }.join(" ") }
   end
 
-  def get_system_name
-    bracketed_name = find_section('SYSTEM NAME')
-    if bracketed_name
-      @system_name = bracketed_name.join(', ')
-      @system_name = @system_name.sub ', {}', ''
-      @system_name = @system_name.sub '"]', ''
-      @system_name = @system_name.sub '["', ''
-      @system_name = @system_name.sub '"', ''
-      return @system_name
-    end
+  def cleanup_xml_element_to_string(element)
+    # recursively convert hashes and array down to a string
+    element = xml_hash_to_string(element) if element.class == Saxerator::Builder::HashElement
+    element = xml_array_to_string(element) if element.class.in? [Array, Saxerator::Builder::ArrayElement]
+    element.strip if element.class.in? [String, Saxerator::Builder::StringElement]
   end
 
-  def get_system_number
-    number = find_section('NUMBER')
-    # puts number
-    if number and @system_name
-      parse_system_name_from_number
+  def xml_hash_to_string(element)
+    if element.fetch('P', nil).present?
+      # Grab the paragraphs out of any hashes
+      cleanup_xml_element_to_string(element.fetch('P', nil))
+    else
+      # A very few section headers have a hash with E
+      cleanup_xml_element_to_string(element.fetch('E', nil))
     end
   end
 
+  def xml_array_to_string(element)
+    # Arrays can contain hashes and arrays
+    # turn all the inside elements into strings then join on spaces
+    element.map do |e|
+      cleanup_xml_element_to_string(e)
+    end.join(" ")
+  end
+
+  def unwanted_parts_of_suplinf(node)
```

Comment:
It's a tag in the SORN XML. https://github.com/18F/all_sorns/blob/main/spec/fixtures/files/sorn.xml#L50

It's a weird one, because it contains the PRIACT tag, where the majority of the content of the SORN lives. So parsing it required some custom logic.